charts/pulsar/templates/function-worker/function-worker-statefulset.yaml

```diff
@@ -0,0 +1,204 @@
+        prometheus.io/port: "{{ .Values.functions.ports.http }}"
+        {{- if .Values.functions.autoRollDeployment }}
+        checksum/config: {{ include (print $.Template.BasePath "/broker/broker-configmap.yaml") . | sha256sum }}
+    {{- end }}
+    {{- with .Values.functions.annotations }}
+    {{ toYaml . | indent 8 }}
+    {{- end }}
+    spec:
+      {{- if .Values.imagePullSecrets }}
+      imagePullSecrets:
+        - name: {{ .Values.imagePullSecrets }}
+      {{- end }}
+      securityContext:
+      {{- with .Values.broker.securityContext }}
+      {{ toYaml . | indent 8 }}
+      {{- end }}
+      {{- if or .Values.broker.serviceAccount.use (and .Values.components.functions  (not .Values.functions.useDedicateRunner)) }}
+      serviceAccountName: {{ template "pulsar.function.serviceAccount" . }}
+      {{- end }}
+      {{- if .Values.functions.nodeSelector }}
+      nodeSelector:
+      {{ toYaml .Values.functions.nodeSelector | indent 8 }}
+      {{- end }}
+      {{- if .Values.functions.tolerations }}
+      tolerations:
+      {{ toYaml .Values.functions.tolerations | indent 8 }}
+      {{- end }}
+      affinity:
+        {{- if and .Values.affinity.anti_affinity .Values.broker.affinity.anti_affinity}}
+        podAntiAffinity:
+          {{ .Values.functions.affinity.type }}:
+          {{ if eq .Values.functions.affinity.type "requiredDuringSchedulingIgnoredDuringExecution"}}
+          - labelSelector:
+              matchExpressions:
+                - key: "app"
+                  operator: In
+                  values:
+                    - "{{ template "pulsar.name" . }}"
+                - key: "release"
+                  operator: In
+                  values:
+                    - {{ .Release.Name }}
+                - key: "component"
+                  operator: In
+                  values:
+                    - {{ .Values.functions.component }}
+            topologyKey: "kubernetes.io/hostname"
+              {{ else }}
+              - weight: 100
+                podAffinityTerm:
+                  labelSelector:
+                    matchExpressions:
+                      - key: "app"
+                        operator: In
+                        values:
+                          - "{{ template "pulsar.name" . }}"
+                      - key: "release"
+                        operator: In
+                        values:
+                          - {{ .Release.Name }}
+                      - key: "component"
+                        operator: In
+                        values:
+                          - {{ .Values.functions.component }}
+                  topologyKey: "kubernetes.io/hostname"
+        {{ end }}
+      {{- end }}
+      terminationGracePeriodSeconds: {{ .Values.broker.gracePeriod }}
+      initContainers:
+        # This init container will wait for zookeeper to be ready before
+        # deploying the bookies
+        - name: wait-zookeeper-ready
```

Comment:
function worker doesn't talk to zk directly, so it doesn't need to wait for zk ready IMO.

Reply:
the init containers are actually copied from proxy, I assume proxy and function worker both depend on broker so should have same prerequisite, if function worker doesn't need to wait for zk then I'll remove this one.